validators.py

```
@@ -0,0 +1,75 @@
+    # Validate nation against cache
+    if not CalendarMetadataCache.is_valid_national(nation):
+        available = CalendarMetadataCache.get_national_calendars()
+        return f"❌ National calendar not found for: {nation}\n💡 Available nations: {', '.join(available)}"
+
+    return nation.strip().upper()
+
+
+def validate_diocese(diocese: str) -> str:
+    """Validate and normalize diocese ID."""
+    if not diocese.strip():
+        raise ValueError("Diocese ID is required")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)